packages/components/src/slot-fill/index.tsx

```diff
@@ -27,29 +39,39 @@ export function Fill( props ) {
 		</>
 	);
 }
-export const Slot = forwardRef( ( { bubblesVirtually, ...props }, ref ) => {
+
+export function UnforwardedSlot(
+	props: WordPressComponentProps< SlotComponentProps, 'div' >,
+	ref: ForwardedRef< any >
+) {
+	const { bubblesVirtually, ...restProps } = props;
```

Comment:
This destructuring could be done during function definition, too.

Reply:
@tyxla I modified it this way referring to the migration guide, but should I do it in the function definition?

https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#refactoring-a-component-to-typescript:~:text=If%20the%20components%20in%20the%20ignored%20files%20are%20destructuring%20props%20directly%20from%20the%20function%27s%20arguments%2C%20move%20the%20props%20destructuring%20to%20the%20function%27s%20body%20(this%20is%20to%20avoid%20TypeScript%20errors%20from%20trying%20to%20infer%20the%20props%20type)%3A

Reply:
My understanding is that this is done only when type inference doesn't work as expected, but given that this is specifically mentioned in the migration guide, feel free to leave it as is 👍